openslides_backend/presenter/check_mediafile_id.py

```diff
@@ -38,30 +43,71 @@ class CheckMediafileId(BasePresenter):
     schema = check_mediafile_id_schema
 
     def get_result(self) -> Any:
-        mediafile = self.datastore.get(
-            FullQualifiedId(Mediafile.collection, self.data["mediafile_id"]),
-            mapped_fields=["filename", "is_directory"],
-        )
+        try:
+            mediafile = self.datastore.get(
+                FullQualifiedId(Mediafile.collection, self.data["mediafile_id"]),
+                mapped_fields=[
+                    "filename",
+                    "is_directory",
+                    "owner_id",
+                    "token",
+                    "mimetype",
+                ],
+            )
+        except DatastoreException:
+            return {"ok": False}
+        if not mediafile.get("owner_id") or mediafile.get("is_directory"):
+            return {"ok": False}
+        self.check_permissions()
+        collection, _ = mediafile["owner_id"].split(KEYSEPARATOR)
+        if collection == "organization":
+            return self.get_organization_result(mediafile)
+        elif collection == "meeting":
+            return self.get_meeting_result(mediafile)
+        return {"ok": False}
 
-        if not mediafile or mediafile.get("is_directory"):
+    def get_organization_result(self, mediafile: Dict[str, Any]) -> Any:
+        if not mediafile.get("token"):
+            return {"ok": True, "filename": mediafile.get("filename")}
+        if not mediafile.get("mimetype"):
             return {"ok": False}
         self.check_permissions()
+        extension = mimetypes.guess_extension(mediafile["mimetype"])
+        if extension is None:
+            return {"ok": False}
+        filename = mediafile["token"] + extension
+
+        return {"ok": True, "filename": filename}
 
+    def get_meeting_result(self, mediafile: Dict[str, Any]) -> Any:
+        if not mediafile.get("filename"):
+            return {"ok": False}
         return {"ok": True, "filename": mediafile["filename"]}
```

Comment:
Check the description in https://github.com/OpenSlides/OpenSlides/wiki/check_mediafile_id again: You should be able to unify these methods, the only exception is if a token is set. Also, you are currently calling `check_permissions` twice to organization mediafiles.

Reply:
Ah, okay, I have unified the two methods.